lib/utils/report.mjs

```diff
@@ -16,54 +16,45 @@ import { isNumber } from './validateTypes.mjs';
  * @type {import('stylelint').Utils['report']}
  */
 export default function report(problem) {
-	const { ruleName, result, message, messageArgs, line, node, index, endIndex, word, severity } =
-		problem;
-
-	if (isFixApplied(problem)) return;
-
-	const ruleSeverityOption = severity || result.stylelint.ruleSeverities[ruleName];
-	const defaultSeverity = result.stylelint.config?.defaultSeverity || 'error';
-	const ruleSeverity =
-		typeof ruleSeverityOption === 'function'
-			? ruleSeverityOption(...(messageArgs || [])) || defaultSeverity
-			: ruleSeverityOption;
+	const { node, index, endIndex, line, start, end, result, ruleName, word, ...rest } = problem;
+	const {
+		disabledRanges,
+		quiet,
+		ruleSeverities,
+		config: { defaultSeverity = 'error', ignoreDisables } = {},
+		customMessages: { [ruleName]: message = rest.message } = {},
+		customUrls: { [ruleName]: customUrl } = {},
+	} = result.stylelint;
+	const { messageArgs = [], severity = ruleSeverities[ruleName] } = rest;
+	const ruleSeverity = isFn(severity) ? severity(...messageArgs) || defaultSeverity : severity;
 
 	// In quiet mode, mere warnings are ignored
-	if (result.stylelint.quiet && ruleSeverity !== 'error') {
-		return;
-	}
-
-	// endIndex is optional
-	const { start } = (node && node.rangeBy({ index, endIndex })) || {};
+	if (quiet && ruleSeverity !== 'error') return;
```

Comment:
[suggest] How about extracting the severity literals to constants (`'error'` and `'warning'`)? Put the constants in `lib/constants.mjs` seems good.

```suggestion
	if (quiet && ruleSeverity !== SEVERITY_ERROR) return;
```

Reply:
There are too matches in the code base. Id prefer if it was done in a separate refactor PR. 
i.e. it should be done for all files at once

Reply:
I'm fine with another PR 👍🏼